src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,50 +1,167 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.value = value;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    private T unlink(Node<T> removeNode) {
```

Comment:
your method should be structured by logic, so it's better to move it near remove methods

Reply:
done